src/ray/stats/metric.h

```diff
@@ -191,6 +201,195 @@ struct MetricPoint {
   const opencensus::stats::MeasureDescriptor &measure_descriptor;
 };
 
+enum StatsType : int { COUNT, SUM, GAUGE, HISTOGRAM };
+
+namespace internal {
+void RegisterAsView(opencensus::stats::ViewDescriptor view_descriptor,
+                    const std::vector<opencensus::tags::TagKey> &keys);
+template <StatsType T>
+struct StatsTypeMap {
+  static constexpr const char *val = "_void";
+};
+
+template <>
+struct StatsTypeMap<COUNT> {
+  static opencensus::stats::Aggregation Aggregation(const std::vector<double> &) {
+    return opencensus::stats::Aggregation::Count();
+  }
+  static constexpr const char *val = "_cnt";
+};
+
+template <>
+struct StatsTypeMap<SUM> {
+  static opencensus::stats::Aggregation Aggregation(const std::vector<double> &) {
+    return opencensus::stats::Aggregation::Sum();
+  }
+  static constexpr const char *val = "_sum";
+};
+
+template <>
+struct StatsTypeMap<GAUGE> {
+  static opencensus::stats::Aggregation Aggregation(const std::vector<double> &) {
+    return opencensus::stats::Aggregation::LastValue();
+  }
+  static constexpr const char *val = "_gauge";
+};
+
+template <>
+struct StatsTypeMap<HISTOGRAM> {
+  static opencensus::stats::Aggregation Aggregation(const std::vector<double> &buckets) {
+    return opencensus::stats::Aggregation::Distribution(
+        opencensus::stats::BucketBoundaries::Explicit(buckets));
+  }
+  static constexpr const char *val = "_dist";
+};
+
+template <StatsType T>
+void RegisterView(const std::string &name, const std::string &description,
+                  const std::vector<opencensus::tags::TagKey> &tag_keys,
+                  const std::vector<double> &buckets) {
+  using I = StatsTypeMap<T>;
+  auto view_descriptor = opencensus::stats::ViewDescriptor()
+                             .set_name(name + I::val)
```

Comment:
I'm not sure why we need add extra suffix for every metric. It's confused if users set one metric name but get other different name from exporter.

Reply:
It's for one measure multi-view. It's the only way to do this. It explicitly gives the suffix of the type, I think the user knows what's there. Besides, we already add prefixes there.